builder/remotecontext/lazycontext.go

```diff
@@ -40,14 +40,14 @@ func (c *lazySource) Hash(path string) (string, error) {
 		return "", err
 	}
 
-	fi, err := c.root.Lstat(fullPath)
+	relPath, err := Rel(c.root, fullPath)
 	if err != nil {
-		return "", errors.WithStack(err)
+		return "", errors.WithStack(convertPathError(err, cleanPath))
 	}
 
-	relPath, err := Rel(c.root, fullPath)
+	fi, err := os.Lstat(fullPath)
 	if err != nil {
-		return "", errors.WithStack(convertPathError(err, cleanPath))
+		return relPath, nil
```

Comment:
`if err != nil { return nil }` looks as if it were a typo. Please add a comment so that readers won't get confused?